### PR TITLE
Support the normalize-space Metapath function

### LINKED ISSUE #133: Function `normalize-space()` is not implemented in Metapath processor
### User Story

As a developer of Metaschema-based software, in order to write more efficient and easier to read Metapath in my models, I would like this library to implement the function `normalize-space()`.

### Goals

- [ ] Implement the function for Metapath use
- [ ] Provide positive correct use of the function in unit tests
- [ ] Provide incorrect correct negative use of the function unit tests

### Dependencies

N/A

### Acceptance Criteria

- [ ] All website and readme documentation affected by the changes in this issue have been updated.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### Revisions

_No response_

### PR DESCRIPTION
**Pull request by david-waltermire**
# Committer Notes

- Added support for the normalize-space Metapath function.
- Also fixed a bug in the string literal handling that inconsistently caused leading and trailing quotes to be removed. 
- Also fixed an issue with the build that was causing surefire to inconsistently handle a JUnit extension.

Resolves #133

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/metaschema-framework/metaschema-java/blob/main/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/metaschema-framework/metaschema-java/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you included examples of how to use your new feature(s)?
- [x] Have you updated all website and readme documentation affected by the changes you made?

### CODE REVIEW

File: core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnNormalizeSpace.java
Comment (aj-stein-gsa) [inline core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnNormalizeSpace.java:72]: Non-blocking: you got me to raise my eyebrow, so I decided to read the spec and that's ... interesting. Perhaps add reference to the spec, but not worth blocking the PR over.

```suggestion
    if (focus == null) {
      // Per the specification:
      // If no argument is supplied and the context item is absent]then a dynamic error is raised: [[err:XPDY0002].
      throw new DynamicMetapathException(DynamicMetapathException.DYNAMIC_CONTEXT_ABSENT, "The context is empty");
    }
```

**Review by aj-stein-gsa — DISMISSED** [review]

File: core/src/test/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnNormalizeSpaceTest.java
Comment (aj-stein-gsa) [inline core/src/test/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnNormalizeSpaceTest.java:52]: I will leave this comment outside of the PR review, but I had to read up on the exception hierarchy on Metapath and I am sure it works, but I am not sure how, as a junior maintainer on this project, would have known effectively this is the "right" or "best" class in the exception hierarchy to pick given this bit of code also in the PR. Am I missing something obvious?

https://github.com/metaschema-framework/metaschema-java/pull/163#diff-870b64e7195168003ad5c4706862b2ceac1e3bbdeccc3940c7d279e9e48adadeR69-R72

**Review by aj-stein-gsa — DISMISSED** [review]

File: core/src/test/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnNormalizeSpaceTest.java
Comment (david-waltermire) [reply core/src/test/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnNormalizeSpaceTest.java]: It turns out that this was a bug, which is fixed in the latest commits.

**Review by aj-stein-gsa — APPROVED** [review]
Thanks for discussing the changes around focus, that helped me generally and to understand the additions to the PR. Looks good to me. 🚢 it!

**Pull request merged**
